### PR TITLE
Task 6 (AWS SQS/SNS)

### PR DESCRIPTION
**Pull request by IDiamondDragon**
Frontend:
https://d3jfdstsfa8xga.cloudfront.net/

file for a test lays in the folder 'example'

Evaluation criteria
Reviewers should verify the lambda functions, SQS and SNS topic and subscription in PR.

1 - File **serverless.yml** contains configuration for catalogBatchProcess function - **Done**
2 - File **serverless.yml** contains policies to allow lambda catalogBatchProcess function to interact with SNS and SQS - **Done**
3 - File **serverless.yml** contains configuration for SQS catalogItemsQueue - **Done**
4 - File **serverless.yml** contains configuration for SNS Topic createProductTopic and email subscription - **Done**
Additional (optional) tasks
+1 - catalogBatchProcess lambda is covered by unit tests - **Done**
the test in the file: catalogBatchProcess.test.ts
+1 - set a Filter Policy for SNS createProductTopic in **serverless.yml** (Create an additional email subscription and distribute messages to different emails depending on the filter for any product attribute) - **Done**

**My score: 6 points**

### CODE REVIEW

File: product-service/serverless.ts [hunk 1]
```diff
@@ -219,8 +241,71 @@ const serverlessConfiguration: Serverless = {
...
+          Protocol: 'email',
+          TopicArn: {
+            Ref: 'SNSTopic'
+          },
+          FilterPolicy: {
+            "price": [
+              "10",
+              "2",
+              "4"
+            ]
```
Comment (sleepwalky) [inline product-service/serverless.ts:287]: 🙂

File: product-service/dal/product.service.ts [hunk 2]
```diff
@@ -28,29 +29,74 @@ export class ProductService extends DataBaseService {
...
+                    if (index === products.length - 1) {
+                        await this.disconnect();
+                    }
+                } catch (error) {
+                    console.log("Hasn't added product: ", product);
+                    console.log(error);
+                } 
+            }   
+            )
+        );
```
Comment (sleepwalky) [inline product-service/dal/product.service.ts:100]: It's very suspicious that your DAL knows about sns :shipit:

**Review by sleepwalky — APPROVED** [review]

File: product-service/dal/product.service.ts (hunk 2 above)
Comment (IDiamondDragon) [reply product-service/dal/product.service.ts:100]: I agree with you and I think I will do the refactoring in future. But in this lesson we need to show the work with sns and sqs.

File: product-service/serverless.ts (hunk 1 above)
Comment (IDiamondDragon) [reply product-service/serverless.ts:287]: Just to show the work of the filters for sns. Why don't send some messages with prices 10, 2 and 4 to some email maybe a client asked me to do it.

Comment (sleepwalky) [reply product-service/serverless.ts:287]: image
https://github.com/DzmitryU/rs-aws-be/pull/5/files#diff-bbe99e6bf77e85e88f14980601f1a819286338ad9bad2343f12ab90f8814d2d0
just for learning purposes

**Pull request merged**